### PR TITLE
pkg/me: Add parser for the Intel ME Flash Partition Table

### PR DESCRIPTION
**Pull request by PatrickRudolph**
The flash partition table $FPT describes the paritions found in the ME region.
The new API allows basic detection, enumeration and modification support
for those partitions.

To be used to patch the UEP partition with the KM hash.

Based on Igor Skochinsky talk "Rootkit in your laptop" and ME Analyzer written
by Plato Mavropoulos.

Signed-off-by: Patrick Rudolph <patrick.rudolph@9elements.com>

### CODE REVIEW

File: pkg/me/me.go [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+	ret += fmt.Sprintf(" EntryVersion: 0x%x\n", h.EntryVersion)
+	ret += fmt.Sprintf(" HeaderLength: 0x%x\n", h.HeaderLength)
+	ret += fmt.Sprintf(" HeaderChecksum: 0x%x\n", h.HeaderChecksum)
+	ret += fmt.Sprintf(" TicksToAdd: 0x%x\n", h.TicksToAdd)
+	ret += fmt.Sprintf(" TokensToAdd: 0x%x\n", h.TokensToAdd)
+	ret += fmt.Sprintf(" UMASize: 0x%x\n", h.UMASize)
+	ret += fmt.Sprintf(" Flags: 0x%x\n", h.Flags)
+
+	return ret
+}
```
Comment (ChriMarMe) [inline pkg/me/me.go:39]: Please use strings.Builder with WriteString and/or fmt.Fprintf

File: pkg/me/me.go [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+	ret += fmt.Sprintf(" HeaderLength: 0x%x\n", h.HeaderLength)
+	ret += fmt.Sprintf(" HeaderChecksum: 0x%x\n", h.HeaderChecksum)
+	ret += fmt.Sprintf(" TicksToAdd: 0x%x\n", h.TicksToAdd)
+	ret += fmt.Sprintf(" TokensToAdd: 0x%x\n", h.TokensToAdd)
+	ret += fmt.Sprintf(" UMASizeOrReserved: 0x%x\n", h.UMASizeOrReserved)
+	ret += fmt.Sprintf(" FlashLayoutOrFlags: 0x%x\n", h.FlashLayoutOrFlags)
+	ret += fmt.Sprintf(" Fitc Version: %d.%d.%d.%d\n", h.FitcMajor, h.FitcMinor, h.FitcHotfix, h.FitcBuild)
+
+	return ret
+}
```
Comment (ChriMarMe) [inline pkg/me/me.go:74]: Same. strings.Builder

File: pkg/me/me.go [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+		ret += fmt.Sprintf(" Valid: yes\n")
+	}
+	if e.Flags&1 > 0 {
+		ret += fmt.Sprintf(" Partition: Data\n")
+	} else {
+		ret += fmt.Sprintf(" Partition: Code\n")
+	}
+
+	return ret
+}
```
Comment (ChriMarMe) [inline pkg/me/me.go:111]: This just a list for every entry? Would be better if the result is a proper table.

Also use strings.Builder

Comment (ChriMarMe) [reply pkg/me/me.go:111]: fmt.Fprintf(...) can be very helpful here.

File: pkg/me/me.go (hunk 2 above)
Comment (PatrickRudolph) [reply pkg/me/me.go]: done

File: pkg/me/me.go (hunk 1 above)
Comment (PatrickRudolph) [reply pkg/me/me.go]: done

File: pkg/me/me.go (hunk 3 above)
Comment (PatrickRudolph) [reply pkg/me/me.go]: done

**Review by ChriMarMe — APPROVED** [review]

**Review by zaolin — APPROVED** [review]

**Pull request merged**